packages/java/parser-jvm-plugin-backbone/src/main/java/dev/hilla/parser/plugins/backbone/nodes/TypedNode.java

```diff
@@ -0,0 +1,20 @@
+package dev.hilla.parser.plugins.backbone.nodes;
+
+import java.util.List;
+import java.util.function.UnaryOperator;
+
+import dev.hilla.parser.models.SignatureModel;
+import io.swagger.v3.oas.models.media.Schema;
+import jakarta.annotation.Nonnull;
+
+public interface TypedNode extends AnnotatedNode {
```

Comment:
I wonder if we could have better naming for these nodes. Now it sounds strange because `TypedNode` is not actually typed, it contains a type. And `AnnotatedNode` also contains a model with annotations, not annotated itself. What do you think?

Reply:
I took `AnnotatedModel` as an example for the `AnnotatedNode` interface. The idea is similar: this is something that provides access to annotations and type information.

In practice the node implementation delegates such access to its contained source, but the exact way of delegating is an implementation detail of a particular node class.

Any suggestions for a better name on your mind?

Reply:
After some consideration, I think it is ok to have this naming. If necessary, we could refactor it any time later